### PR TITLE
backported validationDefault() method

### PR DESCRIPTION
**Pull request by bancer**

### CODE REVIEW

File: Model/ShimModel.php [hunk 1]
```diff
@@ -931,4 +940,45 @@ public function alias($alias = null) {
...
+	 * in models.
+	 * {@see https://book.cakephp.org/3.0/en/appendices/orm-migration.html#validation-no-longer-defined-as-a-property}
+	 *
+	 * @param Validator $validator The validator that can be modified to
+	 * add some rules to it.
+	 * @return Validator
+	 * @link https://api.cakephp.org/3.5/class-Cake.Validation.ValidatorAwareTrait.html#_validationDefault
+	 */
+	public function validationDefault(Validator $validator) {
+		Debugger::log($validator);
```
Comment (dereuromark) [inline Model/ShimModel.php]: is that on purpose?

File: Model/ShimModel.php [hunk 2]
```diff
@@ -931,4 +940,45 @@ public function alias($alias = null) {
...
+		return $validator;
+	}
+
+	/**
+	 * Returns an instance of a model validator for this class.
+	 *
+	 * You can implement the method in `validationDefault` in your Model subclass
+	 * should you wish to have the default validation set.
+	 *
+	 * @param ModelValidator $instance Model validator instance.
```
Comment (dereuromark) [inline Model/ShimModel.php]: |null

File: Test/Case/Model/ShimModelTest.php [hunk 3]
```diff
@@ -411,13 +411,44 @@ public function testExistsInvalid() {
...
+		]);
+		$this->assertEquals(5, $actual['User']['id']);
+		$this->assertEquals('John Blow', $actual['User']['user']);
+		$this->assertEquals('12345678', $actual['User']['password']);
+	}
+
+	/**
+	 * Tests validation rules added in `validationDefault()` method.
+	 *
+	 * @return
```
Comment (dereuromark) [inline Test/Case/Model/ShimModelTest.php]: void?

File: Test/Case/Model/ShimModelTest.php [hunk 4]
```diff
@@ -411,13 +411,44 @@ public function testExistsInvalid() {
...
+	 */
+	public function testSaveInvalid() {
+		$actual = $this->User->save([
+			'user' => 'John Blow',
+		]);
+		$this->assertFalse($actual);
+		$this->assertEquals('Please enter the password.', $this->User->validationErrors['password'][0]);
```
Comment (dereuromark) [inline Test/Case/Model/ShimModelTest.php]: "required" here
without translation :)

File: Model/ShimModel.php (hunk 1 above)
Comment (bancer) [reply Model/ShimModel.php]: Nope :)

File: Model/ShimModel.php (hunk 2 above)
Comment (bancer) [reply Model/ShimModel.php]: Are you sure? There is no `|null` [here](https://api.cakephp.org/2.10/source-class-Model.html#3895-3910). This is `@param`, not `@return`.

Comment (dereuromark) [reply Model/ShimModel.php]: Well, your code clearly says `(ModelValidator $instance = null)`

Comment (dereuromark) [reply Model/ShimModel.php]: So null is also allowed.

**Comment by dereuromark** [comment]
NICE WORK!

**Pull request merged**